### PR TITLE
Adding champion scenarios requirement to review guidelines

### PR DESCRIPTION
**Pull request by JonathanGiles**

### CODE REVIEW

File: docs/ReviewProcess.md [hunk 1]
```diff
@@ -13,14 +13,15 @@ Make sure your components follow [Azure SDK Design Guidelines](https://github.co
...
 
 1. Link to the service documentation/specification.
-2. Link to the service REST APIs, if applicable/avaliable
-3. Several code samples showing how the SDK is meant to be used by customers. An example of a good set of usage samples can be found [here](https://github.com/dotnet/corefx/issues/32588).
-4. Listing of the APIs. See below for example and tools to generate it.
-5. If the SDK is already prototyped, dlls/packages/etc with the prototype implementation.
-6. If the SDK already GAed in the past, and this review is for additional APIs, old dlls/packages (to help us understand the changes).
+2. Link to the service REST APIs, if applicable/avaliable.
+3. A listing of 3-5 champion code paths relevant to the SDK component (e.g. upload blob, download blob, list all blobs in a container for blob storage). These identify the critical code pathways that the majority of developers will experience. For each champion code path, a link to a code sample in the repo must be provided. It is expected that these champion code paths are optimized for, ensuring succinct, intuitive, and productive developer experiences are possible for each.
```
Comment (mmacy) [inline docs/ReviewProcess.md]: These can directly inform the examples that should appear in the README. Unsure if it's appropriate to mention that here, but great to see this here.

Comment (JonathanGiles) [reply docs/ReviewProcess.md]: That is a good point - the champion samples should also be the README samples. I think perhaps we might include that in the spec docs guidelines.

File: docs/ReviewProcess.md [hunk 2]
```diff
@@ -13,14 +13,15 @@ Make sure your components follow [Azure SDK Design Guidelines](https://github.co
 
 ## What to Prepare for a Review
 
-To conduct a review, we need the following things from the owners of the SDK:
+To conduct a review, we need the following things from the owners of the SDK component:
 
 1. Link to the service documentation/specification.
-2. Link to the service REST APIs, if applicable/avaliable
-3. Several code samples showing how the SDK is meant to be used by customers. An example of a good set of usage samples can be found [here](https://github.com/dotnet/corefx/issues/32588).
```
Comment (KrzysztofCwalina) [inline docs/ReviewProcess.md]: Your change removed link to example .

File: docs/ReviewProcess.md [hunk 3]
```diff
@@ -13,14 +13,15 @@ Make sure your components follow [Azure SDK Design Guidelines](https://github.co
...
-4. Listing of the APIs. See below for example and tools to generate it.
-5. If the SDK is already prototyped, dlls/packages/etc with the prototype implementation.
-6. If the SDK already GAed in the past, and this review is for additional APIs, old dlls/packages (to help us understand the changes).
+2. Link to the service REST APIs, if applicable/avaliable.
```
Comment (KrzysztofCwalina) [inline docs/ReviewProcess.md]: How is this different from #1 above?

File: docs/ReviewProcess.md (hunk 2 above)
Comment (KrzysztofCwalina) [inline docs/ReviewProcess.md]: Link to example of a smaple was removed.

Comment (JonathanGiles) [reply docs/ReviewProcess.md]: Sample link is still there, it's just under point number 4 now.

File: docs/ReviewProcess.md [hunk 4]
```diff
@@ -13,14 +13,15 @@ Make sure your components follow [Azure SDK Design Guidelines](https://github.co
...
+3. A listing of 3-5 champion code paths relevant to the SDK component (e.g. upload blob, download blob, list all blobs in a container for blob storage). These identify the critical code pathways that the majority of developers will experience. For each champion code path, a link to a code sample in the repo must be provided. It is expected that these champion code paths are optimized for, ensuring succinct, intuitive, and productive developer experiences are possible for each.
+4. Several code samples showing how the SDK is meant to be used by customers. An example of a good set of usage samples can be found [here](https://github.com/dotnet/corefx/issues/32588).
```
Comment (KrzysztofCwalina) [inline docs/ReviewProcess.md]: It's not very clear how the samples differ from the champion code paths, i.e. you have to read the points very carefully to distill the differences.

Comment (JonathanGiles) [reply docs/ReviewProcess.md]: The champions are the critical features of the SDK component that we must get right. The samples are the way in which we can review that these champions are possible in a way that we believe is succinct, intuitive, and productive. Champion samples are the subset of all samples. The goal is to force the identification of the champions, to ensure that all engineering teams know that they will be judged on these primarily, as well as to motivate their thinking around these at design time.

Did I say anything that makes it more clear? :-)

**Comment by JonathanGiles** [comment]
Based on our discussion, I have tweaked the wording, however I have put 'samples' first and 'champions' second, to aid readability. Thoughts?

File: docs/ReviewProcess.md [hunk 5]
```diff
@@ -1,10 +1,10 @@
-We expect all Azure SDKs to go through rigorous APIs reviews similar to those conducted for all .NET APIs.
+We expect all Azure SDK components to go through rigorous APIs reviews similar to those conducted for all .NET APIs.
```
Comment (JeffreyRichter) [inline docs/ReviewProcess.md]: Use "client libraries" instead of "components"

File: docs/ReviewProcess.md [hunk 6]
```diff
@@ -1,10 +1,10 @@
...
 It's critical that the review is conducted early enough to allow time for fixes,
 and sometimes significant API redesign based on the review feedback.
-If you have never not participated in Azure SDK API review,
+If you have never not participated in Azure SDK component API review,
```
Comment (JeffreyRichter) [inline docs/ReviewProcess.md]: Fix grammar and change "component" to "client library"

File: docs/ReviewProcess.md [hunk 7]
```diff
@@ -1,10 +1,10 @@
...
 we recommend that you schedule a pre-review (consulting session) before you start working on the APIs.
 
-Note: Azure SDK Reviews are not REST API (nor swagger) reviews. We review language-specific SDK APIs.
+Note: Azure SDK component reviews are not REST API (nor swagger) reviews. We review language-specific SDK component APIs.
```
Comment (JeffreyRichter) [inline docs/ReviewProcess.md]: component --> client library   (2x)

File: docs/ReviewProcess.md [hunk 8]
```diff
@@ -1,10 +1,10 @@
...
 In particular, we review .NET, Python, Java, and JavaScript APIs, and in rare cases C++, Go, and other language SDKs.
```
Comment (JeffreyRichter) [inline docs/ReviewProcess.md]: Change C++ to C/C++?

File: docs/ReviewProcess.md [hunk 9]
```diff
@@ -13,14 +13,15 @@ Make sure your components follow [Azure SDK Design Guidelines](https://github.co
```
Comment (JeffreyRichter) [inline docs/ReviewProcess.md]: components --> client libraries

File: docs/ReviewProcess.md [hunk 10]
```diff
@@ -13,14 +13,15 @@ Make sure your components follow [Azure SDK Design Guidelines](https://github.co
 
 ## What to Prepare for a Review
 
-To conduct a review, we need the following things from the owners of the SDK:
+To conduct a review, we need the following things from the owners of the SDK component:
```
Comment (JeffreyRichter) [inline docs/ReviewProcess.md]: component --> cl

File: docs/ReviewProcess.md [hunk 11]
```diff
@@ -13,14 +13,15 @@ Make sure your components follow [Azure SDK Design Guidelines](https://github.co
...
 
-1. Link to the service documentation/specification.
-2. Link to the service REST APIs, if applicable/avaliable
-3. Several code samples showing how the SDK is meant to be used by customers. An example of a good set of usage samples can be found [here](https://github.com/dotnet/corefx/issues/32588).
-4. Listing of the APIs. See below for example and tools to generate it.
-5. If the SDK is already prototyped, dlls/packages/etc with the prototype implementation.
-6. If the SDK already GAed in the past, and this review is for additional APIs, old dlls/packages (to help us understand the changes).
+1. Several code samples showing how the SDK is meant to be used by customers. An example of a good set of usage samples can be found [here](https://github.com/dotnet/corefx/issues/32588).
```
Comment (JeffreyRichter) [inline docs/ReviewProcess.md]: SDK --> client library

File: docs/ReviewProcess.md [hunk 12]
```diff
@@ -13,14 +13,15 @@ Make sure your components follow [Azure SDK Design Guidelines](https://github.co
...
+2. A listing of 3-5 champion scenarios relevant to the developer. These must identify the critical scenarios that the majority of developers will experience. For each champion scenario, a link to a code sample in the repo must be provided. It is expected that these champion scenarios are optimized for, ensuring succinct, intuitive, and productive developer experiences are possible for each.
```
Comment (JeffreyRichter) [inline docs/ReviewProcess.md]: Did we settle on champion instead of hero? I thought we agreed on hero.

File: docs/ReviewProcess.md [hunk 13]
```diff
@@ -13,14 +13,15 @@ Make sure your components follow [Azure SDK Design Guidelines](https://github.co
...
+3. Link to the service documentation/specification.
+4. Link to the service REST APIs, if applicable/avaliable.
```
Comment (JeffreyRichter) [inline docs/ReviewProcess.md]: available (typo).

File: docs/ReviewProcess.md [hunk 14]
```diff
@@ -13,14 +13,15 @@ Make sure your components follow [Azure SDK Design Guidelines](https://github.co
...
+5. Listing of the APIs. See below for example and tools to generate it.
+6. If the SDK is already prototyped, dlls/packages/etc with the prototype implementation.
```
Comment (JeffreyRichter) [inline docs/ReviewProcess.md]: SDK --> cl
etc. (missing period)

File: docs/ReviewProcess.md [hunk 15]
```diff
@@ -13,14 +13,15 @@ Make sure your components follow [Azure SDK Design Guidelines](https://github.co
...
+7. If the SDK already GAed in the past, and this review is for additional APIs, old dlls/packages (to help us understand the changes).
```
Comment (JeffreyRichter) [inline docs/ReviewProcess.md]: SDK --> client library

File: docs/ReviewProcess.md (hunk 12 above)
Comment (JonathanGiles) [reply docs/ReviewProcess.md]: I don't think we settled on either as the definitive naming pattern. I'm happy to have it be either.

**Pull request merged**